foundry/pixi/containers/ruler.d.ts

```diff
@@ -1,167 +1,184 @@
-declare interface RulerData {
-  _state: number;
-  class: string;
-  destination: PIXI.Point;
-  name: string;
-  waypoints: [PIXI.Point];
-}
-
 /**
  * The Ruler - used to measure distances and trigger movements
- * @param user - The User for whom to construct the Ruler instance
+ * @param user  - The User for whom to construct the Ruler instance
+ * @param color - The color for the ruler
+ *                (Default value: The user's color or `0x42f4e2`)
  */
 declare class Ruler extends PIXI.Container {
+  constructor(user?: User, { color }?: { color?: number | null });
+
   /**
-   * Track the current measurement state
+   * Record the User which this Ruler references
    */
-  public _state: number;
+  user: User;
 
   /**
-   * Is the Ruler being actively used to measure distance?
+   * The ruler name - used to differentiate between players
    */
-  public readonly active: boolean;
+  name: string;
 
   /**
    * The ruler color - by default the color of the active user
    */
-  public color: number | null;
+  color: number | null;
+
+  /**
+   * This Array tracks individual waypoints along the ruler's measured path.
+   * The first waypoint is always the origin of the route.
+   */
+  waypoints: PIXI.Point[];
 
   /**
    * The current destination point at the end of the measurement
    */
-  public destination: PIXI.Point;
+  destination: PIXI.Point;
+
+  /**
+   * The Ruler element is a Graphics instance which draws the line and points of the measured path
+   */
+  ruler: PIXI.Graphics;
 
   /**
    * The Labels element is a Container of Text elements which label the measured path
    */
-  public labels: PIXI.Container;
+  labels: PIXI.Container;
 
   /**
-   * The ruler name - used to differentiate between players
+   * Track the current measurement state
+   * @see Ruler.STATES
    */
-  public name: string;
+  protected _state: 0 | 1 | 2 | 3;
 
   /**
-   * The Ruler element is a Graphics instance which draws the line and points of the measured path
+   * The possible Ruler measurement states which can occur
    */
-  public ruler: PIXI.Graphics;
+  static STATES: {
```

Comment:
I always added the static properties at the very bottom (because they are assigned after the class definition in foundry.js). Not sure if we have a guideline for this...